lib/blacklight.rb

```diff
@@ -44,24 +47,68 @@ class << self
   # other services (e.g. refworks callback urls)
   mattr_accessor :secret_key
   @@secret_key = nil
-  
+
+  # @deprecated
   def self.solr_file
     "#{::Rails.root.to_s}/config/solr.yml"
   end
-  
+
+  def self.blacklight_config_file
+    "#{::Rails.root.to_s}/config/blacklight.yml"
+  end
+
   def self.add_routes(router, options = {})
     Blacklight::Routes.new(router, options).draw
   end
 
   def self.solr
-    @solr ||=  RSolr.connect(Blacklight.solr_config)
+    Deprecation.warn Blacklight, "Blacklight.solr is deprecated and will be removed in 6.0.0. Use Blacklight::SolrRepository#connection instead", caller
```

Comment:
When you say `Blacklight::SolrRepository#connection`, is #connection a class method, can you literally do `Blacklight::SolrRepository.connection`?  If so, update the message to give you that? If not, I'd need more guidance on where I find a Blacklight::SolrRepository instance to call it's #connection method.